app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,33 @@
+class CommentsController < ApplicationController
+  before_action :get_parent
+
+  def create
+    @comment = @parent.comments.new(comment_params.merge(user: current_user))
+    
+    if @comment.save
+      flash[:notice] = "Comment saved."
+      redirect_to article_path(@comment.parent_post)
+    else
+      flash[:alert] = "Unable to save comment. Please try again."
+      #If there is a better way to write this logic, please tell me. 
+      #Apparently, inline if-else doesn't syntactically work
+      if @parent.is_a?(Article)
+        redirect_to article_path(@parent)
+      else
+        redirect_to article_path(@parent.parent_post)
+      end
```

Comment:
since you asked, there's a couple of different approaches we could take with this. The first would be a new method that determines the article resource. Something like:

```ruby
def article_resource
  @parent.is_a?(Article) ? @parent : @parent.parent_post
end

# redirect_to article_path(article_resource)
```

The other would be an inline ternary, which I think is harder to read, but it would look something like:

```ruby
redirect_to article_path(@parent.is_a?(Article) ? @parent : @parent.parent_post)
```

My vote is for the new `article_resource` method.